pandas/core/arrays/datetimes.py

```diff
@@ -904,7 +881,7 @@ def month_name(self, locale=None):
 
         result = fields.get_date_name_field(values, 'month_name',
                                             locale=locale)
-        result = self._maybe_mask_results(result)
+        result = self._maybe_mask_results(result, fill_value=None)
```

Comment:
this correct?

Reply:
hmm yeah i c. maybe it would be better to infer based on the type of result? I just don't lke having to specific this at all (and here it shoud be np.nan anyhow). IOW it can either be ``iNaT`` if its a datetimelike op, else its ``np.nan`` if the result is object / float.

Reply:
I was thinking of doing something similar in the next pass

Reply:
k cool